### PR TITLE
ssmtp: use the authPassFile option instead of authPass

### PR DESCRIPTION
**Pull request by basvandijk**
This gives users the option of storing the smtp password outside the world-readable Nix store.

This is part of: https://github.com/NixOS/nixpkgs/issues/24288.

I tested this by configuring ssmtp to send mail via my mandrill account:

```nix
{
  networking.defaultMailServer = {
    directDelivery = true;
    root           = "xxx";
    authUser       = "xxx";
    authPassFile   = "xxx"; # I also tested it with: authPass = "xxx";
    hostName       = "smtp.mandrillapp.com";
    useSTARTTLS    = true;
  };
}
```

and then sending some mail.

I also tested it by setting `authPassFile` to a non-existing file or a file with too restrictive permissions. In those cases the correct error message is printed:

> sendmail: Could not open the AuthPassFile xxx

Setting `authPassFile` to an empty file correctly produces the following error message:

> sendmail: Error while reading a line from the AuthPassFile xxx, or it is empty

### CODE REVIEW

File: pkgs/tools/networking/ssmtp/default.nix [hunk 1]
```diff
@@ -10,6 +10,8 @@ stdenv.mkDerivation {
     sha256 = "0dps8s87ag4g3jr6dk88hs9zl46h3790marc5c2qw7l71k4pvhr2";
   };
 
+  patches = [ ./ssmtp_support_AuthPassFile_parameter.patch ];
```
Comment (Mic92) [inline pkgs/tools/networking/ssmtp/default.nix]: Can you add yourself as maintainer, when adding patches likes this?

**Comment by Mic92** [comment]
Your patch looks technically OK. I am only a little worried about maintenance of the patch. Would it not be easier to maintain to have a custom `preStart` section in the service, which reads the configuration from nix store and replace it `@AuthPass@` with the content of the auth pass file stored in `authPassFile`?

File: pkgs/tools/networking/ssmtp/default.nix (hunk 1 above)
Comment (basvandijk) [inline pkgs/tools/networking/ssmtp/default.nix]: Make sense. I just added myself. I will also try to upstream this patch to Debian.

**Comment by basvandijk** [comment]
Any chance this can be merged? I would like to make some progress on #24288.

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**